### ISSUE TITLE
[Regression] Yi 200K models won't load in latest release

### ISSUE THREAD

**Issue by Mihaiii** [huggingface/transformers#29252]
### System Info

- `transformers` version: 4.38.1
- Platform: Linux-5.4.0-167-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.20.3
- Safetensors version: 0.4.2
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.1.0+cu118 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no (but it also doesn't work if I do)
- Using distributed or parallel set-up in script?: no

### Who can help?

@ArthurZucker and @younesbelkada

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Simply try to load a 200K context length Yi model in latest transformers ([v4.38.1](https://github.com/huggingface/transformers/releases/tag/v4.38.1)) and it will crash:

I just retried "01-ai/Yi-6B-200K" to be 100% sure:

```python
model = AutoModelForCausalLM.from_pretrained("01-ai/Yi-6B-200K")
```

Everything is ok when trying to load [01-ai/Yi-6B](https://huggingface.co/01-ai/Yi-6B).
I did not also test [01-ai/Yi-34B-200K](https://huggingface.co/01-ai/Yi-34B-200K), but I tried finetunes of it and they failed so I'm assuming Yi-34B-200K fails too.

This is a regression because loading a 200K model works ok with the release of transformers from ~3weeks ago ([v4.37.2](https://github.com/huggingface/transformers/releases/tag/v4.37.2)).

### Expected behavior

The expected behavior is for the model to load, like it does in previous released versions of the transformers library.

**Comment by ArthurZucker** [comment]
Hey! 
I am running this and so far I do not have any issues. I think you don't have enough memory to run it. That is a regression indeed, and a fix is coming to better cater with caching the `attention_mask`.
TLDR; you don't have enough ram to use 200K anyways (as this would always require creating a 200K causal mask)

**Comment by ArthurZucker** [comment]
This was fixed by #29753

**Comment by ArthurZucker** [comment]
Sorry for the incovenience
